WebApi/WebApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using WebApi.Models.DTO;
+using WebApi.Models.DTO.PostDTOs;
+using WebApi.Services.Services_Interfaces;
+
+
+namespace WebApi.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PostController : ControllerBase
+    {
+        private readonly IPostService _postService;
+        private readonly ILogger<PostController> _logger;
+        public PostController(ILogger<PostController> logger , IPostService postService)
+        {
+            _logger = logger;
+            _postService = postService;
+        }
+
+
+        [HttpGet]
+        public ActionResult<IQueryable<PostDTO>> GetAll([Required][FromHeader] int userID)
+        {
+            var result = _postService.GetAll();
+            if (result != null) return Ok(result);
+            else return NotFound();
+        }
+
+      
+        //[HttpGet("{UserID}")]
+        [HttpGet]
+        public ActionResult<IQueryable<PostDTO>> GetUserPosts([Required][FromQuery] int UserID) //[FromHeader] int headerID
+        {
+            var result = _postService.GetAllOfUser(UserID);
+            if (result != null) return Ok(result);
+            else return NotFound();
+        }
+
+
+        [HttpGet("{postID}")]
+        public ActionResult<PostDTO> Get([Required][FromHeader] int userID, [FromRoute] int postID)
+        {
+            var result = _postService.GetById(postID);
+            if (result != null) return Ok(result);
+            else return NotFound();
+        }
+
+
+        [HttpDelete("{postID}")]
+        public ActionResult<Task> Delete([Required][FromHeader] int userID, [FromRoute] int postID)//, [FromBody] DateTime dateTime) - nie wiem czemu z tym nie działa. Może trzeba zoribć DTO na datetime?
+        {
+            _postService.DeletePostAsync(postID);
+            return Ok();
+        }
+
+        [HttpPut("{postID}")]
+        public ActionResult<Task> Edit([Required][FromHeader] int userID, [FromRoute] int postID, [FromBody] PostEditDTO body)
+        {
+            var result = _postService.EditPostAsync(postID, body);
+            if (result != null) return Ok();
```

Comment:
Beka oczekujesz nulla w razie braku result (czyli nie ma posta o danym id), ale Twoja implementacja w PostRepository rzuca wyjątek, a nie nulla XD

Uwierz mi, nie ma nic bardziej denerwującego niż metody które raz zwracają null a raz rzucają wyjątek -> musimy się zdecydować na jedno i przestrzegać tego

Efekt jest taki, że wywalasz wyjatek, nie łapiesz go i zwraca się ok jakimś cudem przy próbie edycji nie istniejącego posta

Jeżeli zwracasz nulla to niewiadomo co się dzieje, jak zwracasz wyjątek to ten wyjątek mówi co się stało i w zależności czy to wyjątek od braku posta czy jakiś błąd sql to można zwrócic inny response BadRequest() albo NotFound()